blog/content/post/2020-02-05-meta-pcengines-xen.md

```diff
@@ -0,0 +1,140 @@
+---
+title: 'Easy way to stay secure - XEN on the PC Engines apu2'
+abstract: 'Xen Project creates a software system that allows the execution of
+  multiple virtual guest operating systems simultaneously on a single physical machine.
+  In this case, it is a PC Engines apu2 platform.'
+cover: /covers/3mdeb-logo_simple-color.png
+author: norbert.kaminski
+layout: post
+published: false
+date: 2020-02-05
+archives: "2020"
+
+tags:
+  - XEN
+  - meta-pcengines
+  - apu2
+  - PC engines
+categories:
+  - OS Dev
+  - Security
+  - Manufacturing
+
+---
+### What is the Xen Project?
+
+Xen Project creates a software system, that allows the execution of multiple
+virtual guest environments simultaneously on a single physical machine.
+It is valuable when you want to create a complex security software system
+consisting of several operating systems. The unprivileged domains
+(domUs), such as virtual operating systems have no direct access to the
+hardware.
+
+Between software and hardware is a hypervisor. It allocates resources for
+new domains (the virtual operating systems) and schedules the existing ones.
+To learn more about hypervisor see the official
+[documentation](https://wiki.xenproject.org/wiki/Hypervisor).
+
+### What is Domain0?
+
+<i>
+    <h2>
+      <p align=center>Xen Project architecture </p>
+    </h2>
+</i>
+
+![Diagram](https://cloud.3mdeb.com/index.php/s/HNmc8yqbcCQzCje/preview)
+
+As shown in the diagram, Domain0 (Dom0) is the privileged domain and it manages
+unprivileged domains (domUs). Dom0 is started by the hypervisor on boot and
+ensures its usability. Therefore it is the Domain that has direct access to the
+hardware. The Domain0 provides hardware connections to DomUs. Dom0 allows full
+virtualized guests to use the virtual hardware and passes the physical links
+paravirtualized domains.
+
+### How to build Dom0 on the PC Engines apu2?
+
+To build the special domain for the PC Engines apu2 board, We will use the
+[meta-pcengines repository](https://github.com/3mdeb/meta-pcengines/tree/c4ee98ab390b073807173584107c09f49ac1e390).
+The repository contains the yocto layer, that allows you to create a
+minimal image of the dom0. To build a usable version of the dom0 image, the
+meta-pcengines needs a [kas tool](https://kas.readthedocs.io/en/0.19.0/index.html).
```

Comment:
@Asiderr There were already 1.0 and 2.0 kas releases - please use more recent link to docs. Docs is in 1.0 latest I think: https://kas.readthedocs.io/en/1.0/

Reply:
Also, not necessary **needs** - we can as well built it without the kas. Kas is used to manage the yocto layer and provides the docker container for isolation of the development environment.